pdc/apps/compose/views.py

```diff
@@ -1956,3 +1956,45 @@ def destroy(self, request, *args, **kwargs):
             STATUS: 204 NO CONTENT
         """
         return super(ComposeTreeViewSet, self).destroy(request, *args, **kwargs)
+
+
+class ComposeLocationViewSet(viewsets.GenericViewSet):
+    """
+    API endpoint that allows querying compose-variant-arch relevant to location.
+
+    ##Test tools##
+
+    You can use ``curl`` in terminal, with -X GET,
+    or GUI plugins for browsers, such as ``RESTClient``, ``RESTConsole``.
+    """
+    queryset = ComposeTree.objects.all()
+
+    def list(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:composelocations-list:compose_id:location_id:schema$
+
+        __Response__:
+            {
+                "compose_location": string,
+            }
+
+        """
+        compose_trees = ComposeTree.objects.filter(compose__compose_id=kwargs['compose_id'],
+                                                   location__short=kwargs['location_id'],
+                                                   scheme__name=kwargs['schema'])
+        if not compose_trees:
+            result = Response(status=status.HTTP_404_NOT_FOUND,
+                              data={
+                                  'detail': "Can't find compose tree with compose id '%s', location '%s', schema '%s'" %
+                                            (kwargs['compose_id'], kwargs['location_id'], kwargs['schema'])})
+        else:
+            url_set = set(map(lambda obj: obj.url, compose_trees))
```

Comment:
Change to comprehension in 3db97198940d4f17. It's more readable. Thanks.

If return Response directly, it will have 3 exits. For me maybe use `result` to keep it one exit is also acceptable.